dashboard/src/actions/datasetListActions.js

```diff
@@ -40,16 +88,42 @@ export const updateFavoriteRepoNames = (favorites) => async (dispatch) => {
   });
 };
 
-export const updateTblData = (data) => async (dispatch) => {
+export const setPageLimit = (newPerPage) => async (dispatch) => {
   dispatch({
-    type: TYPES.UPDATE_PUBLIC_DATASETS,
-    payload: [...data],
+    type: TYPES.SET_PAGE_LIMIT,
+    payload: newPerPage,
   });
 };
 
-export const callLoading = () => (dispatch) => {
-  dispatch({ type: TYPES.LOADING });
-  setTimeout(() => {
-    dispatch({ type: TYPES.COMPLETED });
-  }, 5000);
+export const setFilterKeys = (startDate, endDate) => async (dispatch) => {
+  dispatch({
+    type: TYPES.SET_DATE_RANGE,
+    payload: { startDate, endDate },
+  });
+};
+
+export const nameFilter = (value) => async (dispatch) => {
+  dispatch({
+    type: TYPES.SET_SEARCH_KEY,
+    payload: value,
+  });
+};
+
+export const applyFilter = () => (dispatch) => {
+  dispatch({
+    type: TYPES.GET_PUBLIC_DATASETS,
+    payload: [],
+  });
+  dispatch({
+    type: TYPES.SET_PAGE_OFFSET,
+    payload: Number(0),
```

Comment:
I thought you were representing the first page as `1`, not `0`? Is this right? Also, do you really need the `Number` constructor for a literal `0`??

Reply:
page is 1 but offset starts from 0. 

Removed the Number constructor